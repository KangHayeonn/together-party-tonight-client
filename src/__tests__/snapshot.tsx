import React from "react";
import { render, screen } from "@testing-library/react";
import Button from "@/components/SampleButton";

describe("Button component", () => {
  test("renders button with correct text", () => {
    const { container } = render(<Button text="Click me" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders disabled button", () => {
    const { container } = render(<Button text="Submit" disabled />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test("renders button with custom class", () => {
    const { container } = render(
      <Button text="Click me" className="custom-button" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
