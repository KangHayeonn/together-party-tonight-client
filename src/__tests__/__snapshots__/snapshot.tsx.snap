// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders button with correct text 1`] = `
<button>
  Click me
</button>
`;

exports[`Button component renders button with custom class 1`] = `
<button
  class="custom-button"
>
  Click me
</button>
`;

exports[`Button component renders disabled button 1`] = `
<button
  disabled=""
>
  Submit
</button>
`;
